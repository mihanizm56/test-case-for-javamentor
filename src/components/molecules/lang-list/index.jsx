import React, { memo } from 'react'
import { LangCard } from '../lang-card'
import './lang-list.css'

export const LangList = memo(({ langData }) => (
    <div className='list-container'>
        {
            langData.map(({name, year, projectsCount, docs, logo}, index) => (
                <LangCard
                    title={name}
                    createdAt={year}
                    numberOfProjects={projectsCount}
                    link={docs}
                    key={index}
                    logo={logo}
                />
            ))
        }
    </div>
))

LangList.defaultProps = { langData: [] }
