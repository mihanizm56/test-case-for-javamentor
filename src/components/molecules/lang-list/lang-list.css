.list-container {
    max-height: 80vh;
    overflow-y: auto;
    padding: 0 15px;
}

.list-container::-webkit-scrollbar-track {
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: #F5F5F5;
}

.list-container::-webkit-scrollbar {
    width: 12px;
    background-color: #F5F5F5;
}

.list-container::-webkit-scrollbar-thumb {
    border-radius: 10px;
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, .3);
    background-color: #555;
}