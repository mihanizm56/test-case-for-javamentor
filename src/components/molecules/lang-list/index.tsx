import React, { memo } from "react";
import { LangCard } from "../lang-card";
import "./lang-list.css";

export interface LangCardType {
    name: string;
    year: number;
    projectsCount: number;
    docs: string;
    logo: string;
}

interface LangListType {
    langData: Array<LangCardType>;
}

export const LangList = memo<LangListType>(({ langData }) => (
    <div className="list-container">
        {
            langData.map(({name, year, projectsCount, docs, logo}, index) => (
                <LangCard
                    title={name}
                    createdAt={year}
                    numberOfProjects={projectsCount}
                    link={docs}
                    key={index}
                    logo={logo}
                />
            ))
        }
    </div>
));
