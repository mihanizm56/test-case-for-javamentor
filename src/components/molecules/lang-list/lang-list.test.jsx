import React from "react";
import { shallow } from "enzyme";
import { LangList } from "../lang-list";

describe("test LangList", () => {
  const listProps = {
    langData: [
      {
        name: "test_name",
        year: 1,
        projectsCount: 1,
        docs: "test_docs",
        logo: "test_logo"
      },
      {
        name: "test_name",
        year: 1,
        projectsCount: 1,
        docs: "test_docs",
        logo: "test_logo"
      },
      {
        name: "test_name",
        year: 1,
        projectsCount: 1,
        docs: "test_docs",
        logo: "test_logo"
      }
    ]
  };
  let component;

  beforeEach(() => {
    jest.clearAllMocks();
    component = shallow(<LangList {...listProps} />);
  });

  describe("test render contract", () => {
    it("renders html-content", () => {
      expect(component).toMatchSnapshot();
    });
  });
  describe("test props contract", () => {
    it("check number of LangCard components", () => {
      expect(component.find("Memo()").length).toEqual(3);
    });

    it("check", () => {
      const specialProps = {
        langData: [
          {
            name: "test_name",
            year: 1,
            projectsCount: 1,
            docs: "test_docs",
            logo: "test_logo"
          }
        ]
      };
      const specialComponent = shallow(<LangList {...specialProps} />);
      expect(specialComponent.find("Memo()").props()).toEqual({
        createdAt: 1,
        link: "test_docs",
        logo: "test_logo",
        numberOfProjects: 1,
        title: "test_name"
      });
    });
  });
});
