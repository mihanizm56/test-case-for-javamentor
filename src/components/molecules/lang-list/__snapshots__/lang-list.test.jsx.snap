// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LangList test render contract renders html-content 1`] = `
<div
  className="list-container"
>
  <Memo()
    createdAt={1}
    key="0"
    link="test_docs"
    logo="test_logo"
    numberOfProjects={1}
    title="test_name"
  />
  <Memo()
    createdAt={1}
    key="1"
    link="test_docs"
    logo="test_logo"
    numberOfProjects={1}
    title="test_name"
  />
  <Memo()
    createdAt={1}
    key="2"
    link="test_docs"
    logo="test_logo"
    numberOfProjects={1}
    title="test_name"
  />
</div>
`;
